align/main.py

```diff
@@ -252,7 +252,12 @@ def schematic2layout(netlist_dir, pdk_dir, netlist_file=None, subckt=None, worki
         primitive_dir.mkdir(exist_ok=True)
         for block_name, block_args in primitives.items():
             logger.debug(f"Generating primitive: {block_name}")
-            generate_primitive(block_name, **block_args, pdkdir=pdk_dir, outputdir=primitive_dir)
+            uc = generate_primitive(block_name, **block_args, pdkdir=pdk_dir, outputdir=primitive_dir)
+            if hasattr(uc, 'metadata'):
+                primitives[block_name]['metadata'] = copy.deepcopy(uc.metadata)
+        
+        with (topology_dir / '__primitives__.json').open( 'wt') as fp:
```

Comment:
@soneryaldiz We shouldn't be modifying something in 1_topology when we run the 2_primitive flow. Do we need to make a copy of __primitives__.json and put it in the 2_primitives directory (maybe changing the name of it so there are no misunderstandings.

Reply:
Sounds good. I will do right away.